compiler/rustc_target/src/abi/mod.rs

```diff
@@ -1495,9 +1495,11 @@ impl<'a, Ty> TyAndLayout<'a, Ty> {
     ///
     /// This code is intentionally conservative, and will not detect
     /// * zero init of an enum whose 0 variant does not allow zero initialization
```

Comment:
Since these are aggregate types for LLVM, I don't think we ever trigger LLVM UB in any of these cases. We do check that 0x01-filling is a valid discriminant (based on the validity range of the discriminant field), and we don't tell LLVM about the variants until one is actually accessed.